### PR TITLE
Acceptance test refractoring

### PR DESCRIPTION
**Pull request by tshihad**
## Changes
- Added README for acceptance test framework which can be followed to create new test suites
- Refractored existing framework.
- Remove `panic` on error and replaced with `t.Fatalf()`
- `atf` framework will now supports `expect_error`.
- Added new random number creator
- Added new test case for instance with `expect_error`

### CODE REVIEW

File: acc-testcases/README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+			return iClient.GetSpecificMyResource(getAccContext(), id)
+		},
+	}
+
+	acc.RunDataSourceTests(t)
+}
+```
+
+`GetAPI` function should contains Get Sepecific API for the a resource/Data source. This is
```
Comment (manjunath-batakurki) [inline acc-testcases/README.md:74]: Specific

File: acc-testcases/README.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+
+## Writing an Acceptance Test
+
+Acceptance test should be written using vmaas acceptance test framework or `atf`. This
+framework supports writing test suites for both resources and data sources.
+
+### Writing Acceptance Test for New Resource/Datasource
+
+Create new test file wil the format `<resource/data_source>_<name of the resource/data source>_test.go`.
```
Comment (manjunath-batakurki) [inline acc-testcases/README.md:41]: an example of the same format would help.

File: acc-testcases/README.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+			return iClient.GetSpecificMyResource(getAccContext(), id)
+		},
+	}
+
+	acc.RunDataSourceTests(t)
+}
+```
+
+`GetAPI` function should contains Get Sepecific API for the a resource/Data source. This is
+used for validation for that specific resource/data source.
```
Comment (manjunath-batakurki) [inline acc-testcases/README.md:75]: of

File: acc-testcases/README.md (hunk 1 above)
Comment (tshihad) [reply acc-testcases/README.md]: Done

File: acc-testcases/README.md (hunk 2 above)
Comment (tshihad) [reply acc-testcases/README.md]: Done

File: acc-testcases/README.md (hunk 3 above)
Comment (tshihad) [reply acc-testcases/README.md]: Done

**Review by gandharvas — APPROVED** [review]

**Review by manjunath-batakurki — APPROVED** [review]

**Pull request merged**